mlos_core/mlos_core/optimizers/observations.py

```diff
@@ -0,0 +1,361 @@
+#
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+#
+"""Contains classes for observations and suggestions."""
+
+from __future__ import annotations
```

Comment:
Unnecesssary?

Reply:
Maybe you "just" need to use the quoted form of class names when giving type hints?

e.g.,
```python
def get_suggestion(...) -> "Suggestion":
   ...
```

Reply:
You can also do something like the following if necessary:

```python
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from mlos_core.optimizers.data_types.suggestion import Suggestion
```

But I think the quoted version + the internal import like I described above should be fine.